### ISSUE TITLE
Create new CLI command: enable/disable Magento Profiler

### ISSUE THREAD

**Issue by vrann** [magento/magento2#9277]
In order to help developers to work with the Magento, create new commands which will simplify the debugging of the application. Currently, enabling of these setting require doing that either with the UI or changing the configuration of the di.xml.

Create CLI command which will make possible to enable/disable Magento Profiler

**Comment by okorshenko** [comment]
Hi @SasiKiranK please let us know if you are still working in this issue. Thank you

**Comment by osrecio** [comment]
Can I help with this task?

**Comment by okorshenko** [comment]
Hi @osrecio Sure! I will assign it to you. Thank you

**Comment by osrecio** [comment]
Hi @vrann and @okorshenko , I have question about this task:

In file: [app/bootstrap.php](https://github.com/magento/magento2/blob/develop/app/bootstrap.php#L52) `$_SERVER['MAGE_PROFILER']` is not empty. 
The problem of this that you can do:`SetEnv MAGE_PROFILER "html"` in your `.htaccess` but you can have another HTTP Server like `nginx`.

Can I move this Enviroment Variable to env.php like `MAGE_MODE` ?

I think is better approach, I can keep 2 ways from `$_SERVER` and from `env.php` What do you think?

Thanks!

**Comment by peterjaap** [comment]
@osrecio I don't think it's a good idea to move it to `env.php`. The reason being that `env.php` is not yet parsed by `app/bootstrap.php` by the time it now initializes the profiler, thereby eliminating the possibility to profile everything between `app/bootstrap.php` and the moment it parses `env.php`.

My suggestion would be to use a stopfile (like `./var/profiler`) to activate the profiler, and use the CLI command to add/remove it and set the profiler type (`html`, `csv` or something custom).

Quick write-out (I'll be happy to create a PR for it);

Instead of;
```
if (!empty($_SERVER['MAGE_PROFILER'])
    && isset($_SERVER['HTTP_ACCEPT'])
    && strpos($_SERVER['HTTP_ACCEPT'], 'text/html') !== false
) {
    \Magento\Framework\Profiler::applyConfig(
        $_SERVER['MAGE_PROFILER'],
        BP,
        !empty($_SERVER['HTTP_X_REQUESTED_WITH']) && $_SERVER['HTTP_X_REQUESTED_WITH'] == 'XMLHttpRequest'
    );
}
```

We use;

```
if (
    (!empty($_SERVER['MAGE_PROFILER']) || file_exists(BP . '/var/profiler'))
    && isset($_SERVER['HTTP_ACCEPT'])
    && strpos($_SERVER['HTTP_ACCEPT'], 'text/html') !== false
) {
    \Magento\Framework\Profiler::applyConfig(
        strlen($_SERVER['MAGE_PROFILER']) ? $_SERVER['MAGE_PROFILER'] : file_get_contents(BP . '/var/profiler'),
        BP,
        !empty($_SERVER['HTTP_X_REQUESTED_WITH']) && $_SERVER['HTTP_X_REQUESTED_WITH'] == 'XMLHttpRequest'
    );
}
```

And then use `php bin/magento dev:profiler:enable <type>` (with type default `html`) and `php bin/magento dev:profiler:disable` to enable and disable the profiler.

**Comment by osrecio** [comment]
GO ahead with PR, I forgot this task :) . I think your idea is good, I like it like maintenance file, etc

Thanks @peterjaap

**Comment by peterjaap** [comment]
@osrecio 45 minutes booooom :fire:

**Comment by osrecio** [comment]
You’re the best 😎

**Comment by okorshenko** [comment]
Delivered to 2.2-develop branch.
The feature will be a part of 2.2.3

**Comment by keharper** [comment]
@peterjaap I need to know more about these commands so that they can be documented. 

* Are they intended to replace the functionality described at http://devdocs.magento.com/guides/v2.2/config-guide/bootstrap/mage-profiler.html ?

* What happens if the profiler is already enabled and you run the `enable` command?

* Can the `enable` command take any arguments other than `html` or `csvfile`?

**Comment by peterjaap** [comment]
@keharper 
1. It does not replace it per se, it is just another way to enable/disable the profiler. Also, this new command does not enable you to enable the dependency graph option.
2. It will just enable it again (/still); https://github.com/magento/magento2/pull/11407/files#diff-30813cda1ee05b4cd04f143198b6d5e1R90
3. No. It will give a warning in that case; https://github.com/magento/magento2/pull/11407/files#diff-30813cda1ee05b4cd04f143198b6d5e1R85

**Comment by magento-engcom-team** [comment]
Hi @vrann. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1359 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 1db3f8f6a336026d4a4493c8ba8d3e165ade66d8
 - 412d2287ee5a1a9620fb2a256bab562bdd10fe49
 - 02620b09456d77408ccbbd6a6b61a7b7ebcf73fa
 - 57c5d8a819e403905bd4d5419564d184712bc142
 - 94a3f0346ea086626d8b11c08fd397b11a570caa
 - 0d557dc99c15c6638e1c894ce46cb681a74b8912
 - 48baf590f86590c44bba81a9d5303902aed9d8d7
 - dc3bb78962c4522eb5df8a1a8f40c3bbe2b623fc
 - 0ab45334ef4d44e55c9f84f9c6e4b6a9e08e9237
 - 00f64b077ce00e55216e6ea9fb39071dc74b51a1
 - 440a1b5f9e6309fa87790a5f5b7d1b8fd7ffa53d
 - 4ed8f5b65d5d1779522653102120917c5f9578be
 - 656eeb5587eb1baebaffdc8ba68f1028afda8910
 - fd2f731f7efc4a2cec9e03f8de88bf39c0988829
 - 2c1a23faa60805bd63ff62216309d57a052612d1
 - f7ff71a92bcf56679b82c22549244e1adb9f4ed2
 - ac9bcfa15b1b0b98f9cb3851f4d9b56e18ede89b
 - 55debe0c2b514e37271ed6edf20b95fec6cc41c9
 - 349508628dc1c4334e13b3541227505e26c93c64
 - d678f1568b6e2d30c6d651dd3da02e0f3512353e
 - bf8ee3089f005e267f0b14cf81b14e9a3ee26986
 - 335c91c07dde6f90309f11cf0efcbd9127630955
 - c5a823f95161f7c13fb0947abb445fdc9e9a620a
 - 696a5a575513bb87c47ffc15a571034b0b91d7ae

The fix will be available with the upcoming 2.3.0 release.
